### DISCUSSION TITLE
Running into a compilation error when trying to import agent-rs crate. Not seeing the same error from a barebones Axum project. Thoughts?

### DISCUSSION THREAD

**Discussion by saikatdas0790** [leptos-rs/leptos#2022] [Q&A] [answered]
I am trying to use this crate in an SSR'd cargo-leptos project:
https://crates.io/crates/ic-agent

It seems to work fine with a barebones axum project. Here is a minimal reproduction:
https://github.com/saikatdas0790/minimal_reproduction_import_agent_rs_in_axum_project

Try cloning and `cargo run`. It boots without any errors.

But if doing the same import in a `cargo-leptos` project, running into the following errors.
![image](https://github.com/leptos-rs/leptos/assets/7412443/da9ae95a-e9c5-4894-a4c3-4fb8bfdf033d)

Minimal reproduction here:
https://github.com/saikatdas0790/minimal-reproduction-add-agent-rs-to-cargo-leptos-project

I'm unsure if the issue is with the crate or `cargo-leptos`.

Thoughts?

**Comment by gbj** [comment]
Is it failing while trying to compile it for WASM? i.e,. have you tried [making it optional and only enabling it for SSR](https://leptos-rs.github.io/leptos/ssr/24_hydration_bugs.html?highlight=optional%20%3D%20true#not-all-server-code-can-run-on-the-client)?

**Reply by saikatdas0790** [reply]
That's the thing. I haven't imported this crate into any code yet with a `use` statement.

This is simply after adding it as a dependency and running `cargo-leptos watch`.

Let me try that though.

**Reply by gbj** [reply]
Correct. `cargo` builds all your dependencies for any given target before it begins building your application code, not after it detects a `use` statement in your application code.

**Reply by saikatdas0790** [reply, marked as answer]
This was solved by setting a feature flag in the imported crate.

More details [here](https://forum.dfinity.org/t/bug-compilation-errors-when-trying-to-add-ic-agent-to-an-axum-project-error-attached/24671/4?u=saikatdas0790)
